### PR TITLE
Add DEPLOYMENT.md

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description

Part of my ongoing work to improve documentation for our infrastructure.

## Testing Instructions

Read the new document and let me know if it needs more information or if anything doesn't make sense or is unclear.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [N/A] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by obulat — APPROVED** [review]
The document is well-written, thank you for documenting every step, @sarayourfriend!

> We need a GitHub workflow that runs the Playwright tests against the staging deployment automatically after staging is deployed. This will require some Playwright configuration tweaks and increased test timeouts due to network latency.

Would we use the recorded tapes when testing staging? I guess we would need to be running the talkback server together with the staging then.

File: DEPLOYMENT.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+1. We need a GitHub workflow that runs the Playwright tests against the staging deployment automatically after staging is deployed. This will require some Playwright configuration tweaks and increased test timeouts due to network latency.
+2. We would need to create a queue so that new staging deployments do not occur until the current staging deployment is live on production. This could be implemented by polling the `/version.json` endpoints of both environments and only liberating a new staging deployment for new merges once the endpoints both report the same `release` property.
+3. We would need to update the production deployment workflow to not create a new Docker build and instead just pull the current tag from the staging `/version.json` endpoint.
+
+Note that this workflow does make the process of merging a PR heavier than it is today. This comes with some benefits:
+
+1. PRs will likely be more carefully tested as the consequence of merging them is higher and more immediately consequential.
+2. Bugs will be fixed in production more quickly.
```
Comment (krysal) [inline DEPLOYMENT.md:78]: This **Future end-to-end test integration** is an exciting plan ✨ and I immediately thought about these points while reading. When the time comes, we would need to really compromise in not merging by pure urgency and include tests with the changes, or I guess we can also define very clearly the exceptions to these constraints.

**Review by krysal — APPROVED** [review]
Thanks for writing this document, it's quite help helpful.

File: DEPLOYMENT.md (hunk 1 above)
Comment (sarayourfriend) [reply DEPLOYMENT.md:78]: I'm interested in ways in which we can hold ourselves accountable for better testing. Coverage data could be one way but I'm sure there are other creative ones.

**Pull request merged**